src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -1032,6 +1118,24 @@ skip:
             bool parent_requested = m_block_queue.requested(new_block.prev_id);
             if (!parent_requested)
             {
+              // we might be able to ask for that block directly, as we now can request out of order,
+              // otherwise we continue out of order, unless this block is the one we need, in which
+              // case we request block hashes, though it might be safer to disconnect ?
+              if (start_height > previous_height)
```

Comment:
I had this question because of this comment

    // as we now can request out of order,

I interpreted it as meaning that the new pruning scheme now allows us to request blocks out of order, or conversely, out-of-order request is not allowed when pruning is disabled. My understanding on how block queue and download take place is generally quite limited, so probably my question is irrelevant.